code/modules/events/event_dynamic.dm

```diff
@@ -45,45 +45,128 @@ var/list/event_last_fired = list()
-
 	possibleEvents[/datum/event/pda_spam] = max(min(25, player_list.len) * 4, 200)
-	possibleEvents[/datum/event/money_lotto] = max(min(5, player_list.len), 50)
+	possibleEvents[/datum/event/money_lotto] = 10
+	possibleEvents[/datum/event/brand_intelligence] = 10 + 10 * active_with_role["Janitor"] +  5 * active_with_role["Engineer"]
 	if(account_hack_attempted)
-		possibleEvents[/datum/event/money_hacker] = max(min(25, player_list.len) * 4, 200)
+		//Made account hack far rarer, reasoning provided in a forum thread on 2016-06-28
+		//In short, its meaningless, annoying, and it trains people to ignore request consoles
+		//Consider restoring it to commonality when economy is fleshed out and currency becomes meaningful
+		possibleEvents[/datum/event/money_hacker] = 10
 
+	possibleEvents[/datum/event/infestation] = 50 + 50 * active_with_role["Janitor"]
 
-	possibleEvents[/datum/event/carp_migration] = 20 + 10 * active_with_role["Engineer"]
-	possibleEvents[/datum/event/brand_intelligence] = 20 + 25 * active_with_role["Janitor"]
+	//Wallrot made much less engineer-dependant, but more gardener-dependant.
+	possibleEvents[/datum/event/wallrot] = 75 + 5 * active_with_role["Engineer"] + 20 * active_with_role["Gardener"]
 
-	possibleEvents[/datum/event/rogue_drone] = 5 + 25 * active_with_role["Engineer"] + 25 * active_with_role["Security"]
-	possibleEvents[/datum/event/infestation] = 100 + 100 * active_with_role["Janitor"]
 
-	possibleEvents[/datum/event/communications_blackout] = 50 + 25 * active_with_role["AI"] + active_with_role["Scientist"] * 25
-	possibleEvents[/datum/event/ionstorm] = active_with_role["AI"] * 25 + active_with_role["Cyborg"] * 25 + active_with_role["Engineer"] * 10 + active_with_role["Scientist"] * 5
-	possibleEvents[/datum/event/grid_check] = 25 + 10 * active_with_role["Engineer"]
-	possibleEvents[/datum/event/electrical_storm] = 15 * active_with_role["Janitor"] + 5 * active_with_role["Engineer"]
-	possibleEvents[/datum/event/wallrot] = 30 * active_with_role["Engineer"] + 50 * active_with_role["Gardener"]
 
-	if(!spacevines_spawned)
-		possibleEvents[/datum/event/spacevine] = 10 + 5 * active_with_role["Engineer"]
-	if(minutes_passed >= 30) // Give engineers time to set up engine
-		possibleEvents[/datum/event/meteor_wave] = 10 * active_with_role["Engineer"]
-		possibleEvents[/datum/event/meteor_shower] = 20 * active_with_role["Engineer"]
-		possibleEvents[/datum/event/blob] = 10 * active_with_role["Engineer"]
+
+
+
+	//Moderate Events
+	//================================================
+
+		//Combat oriented events, more heavily weighted to number of sec officers
+	possibleEvents[/datum/event/rogue_drone] = 50 + 25 * active_with_role["Security"]
+	possibleEvents[/datum/event/spider_infestation] = 50 + 25 * active_with_role["Security"]
+	possibleEvents[/datum/event/carp_migration] = 50 + 25 * active_with_role["Security"]
+
+	possibleEvents[/datum/event/communications_blackout] = 40//Comms blackout is too common and annoying
```

Comment:
Communication blackout is the least damaging of the random events and force people to use the not-often used wall radios. I'd say they don't happen enough actually.